### ISSUE TITLE
Failure to load certain key names

### ISSUE THREAD

**Issue by spacemud** [Southclaws/samp-ini#2]
I'm having trouble reading values with certain key names. Here's the original file that I tried to read:
```ini
server=irc.gtanet.com
port=6667
```
And the code to test it:
```pawn
#include <ini>

main()
{
    ini_open("test.ini");

    new
        key1[16],
        key2,
        error;

    error = ini_getString("key1", key1);
    printf("key1: %s error: %d", key1, error);

    error = ini_getInt("key2", key2);
    printf("key2: %d error: %d", key2, error);

    ini_close();

    return 1;
}
```
Gives:
```
server: irc.gtanet.com error: 0
port: 0 error: 6
```

It seems to read the first key fine, but reading later keys fail.

Renaming the keys to something else works fine, for example changing `server` to `key1` and `port` to `key2` as in the example code above gives:
```
key1: irc.gtanet.com error: 0
key2: 6667 error: 0
```

**Comment by Southclaws** [comment]
Hmm, everything seems to work up until the point of reading...

```log
text="opened existing file" filename="test.ini"
text="read record" record=0 key="server" value="irc.gtanet.com"
text="read record" record=1 key="port" value="6667"
text=""
text="begin ini cache"
text="--record" index=0 deleted=0 key="server" value="irc.gtanet.com"
text="--record" index=1 deleted=0 key="port" value="6667"
text="end ini cache"
key1:  error: 6
key2: 0 error: 6
text="closed file"
```

**Comment by Southclaws** [comment]
Fixed in 1.1.0 - I basically just rewrote the entire thing with BigETI's pawn-map instead of my janky ass binary search tree!
